pkg/util/errors.go

```diff
@@ -281,6 +284,24 @@ func HandleInstanceSharingError(err error, entityName string) error {
 			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, does not match the service offering.", entityName),
 			StatusCode:  http.StatusNotFound,
 		}
+	case ErrMultipleReferenceSelectorResults:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: "Failed to create the reference. Your query selector provided multiple results. Use referenced_instance_id instead.",
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrInvalidReferenceSelectors:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: "Failed to create the reference. Your query selectors are invalid.",
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrNoResultsForReferenceSelector:
+		return &HTTPError{
+			ErrorType:   "NotFound",
+			Description: "Failed to create the reference. Could not find a shared instance with the selectors provided.",
```

Comment:
Couldn't find a shared instance with the provided selectors.

Reply:
Done